usr/src/uts/common/fs/zfs/range_tree.c

```diff
@@ -343,6 +357,15 @@ range_tree_contains(range_tree_t *rt, uint64_t start, uint64_t size)
 }
 
 /*
+ * Same as range_tree_contains, but locates even just a partial overlap.
+ */
+boolean_t
+range_tree_contains_part(range_tree_t *rt, uint64_t start, uint64_t size)
+{
+	return (range_tree_find_impl(rt, start, size) != NULL);
```

Comment:
is the caller supposed to already hold rt_lock? FWIW, https://github.com/openzfs/openzfs/pull/251 (device removal) gets rid of rt_lock.